testutil/ciao-down/instance.go

```diff
@@ -0,0 +1,302 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"net/url"
+	"path"
+	"path/filepath"
+	"strings"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Constants for the Guest image used by ciao-down
+
+const (
+	guestDownloadURL       = "https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img"
+	guestImageFriendlyName = "Ubuntu 16.04"
+	defaultHostname        = "singlevm"
+)
+
+type portMapping struct {
+	Host  int `yaml:"host"`
+	Guest int `yaml:"guest"`
+}
+
+func (p portMapping) String() string {
+	return fmt.Sprintf("%d-%d", p.Host, p.Guest)
+}
+
+type mount struct {
+	Tag           string `yaml:"tag"`
+	SecurityModel string `yaml:"security_model"`
+	Path          string `yaml:"path"`
+}
+
+func (m mount) String() string {
+	return fmt.Sprintf("%s,%s,%s", m.Tag, m.SecurityModel, m.Path)
+}
+
+type instance struct {
+	MemGiB       int           `yaml:"mem_gib"`
+	CPUs         int           `yaml:"cpus"`
+	PortMappings []portMapping `yaml:"ports"`
+	Mounts       []mount       `yaml:"mounts"`
+}
+
+type instanceSpec struct {
+	BaseImageURL  string `yaml:"base_image_url"`
+	BaseImageName string `yaml:"base_image_name"`
+	Hostname      string `yaml:"hostname"`
+}
+
+func newDefaultInstance(ws *workspace, vmType string) *instance {
+	var in instance
+
+	in.Mounts = []mount{
+		{
+			Tag:           "hostgo",
+			SecurityModel: "passthrough",
+			Path:          ws.GoPath,
+		},
+	}
+
+	in.PortMappings = []portMapping{
+		{
+			Host:  10022,
+			Guest: 22,
+		},
+	}
+
+	if vmType == CIAO {
```

Comment:
Actually, the function name is misleading.  It's called newDefaultInstance, but it should be called newLegacyInstance.  I'll rename it and add a comment.